src/NHibernate.Test/NHSpecificTest/NH3488/LinqBulkManipulationFixture.cs

```diff
@@ -0,0 +1,1272 @@
+			}
+
+			using (var s = OpenSession())
+			using (var t = s.BeginTransaction())
+			{
+				int count = s.Query<Joiner>().Where(x => x.JoinedName == "joined-name").Delete();
+				Assert.That(count, Is.EqualTo(1), "Incorrect deletion count on joined class");
+
+				t.Commit();
+				s.Close();
+			}
+		}
+
+		[Test]
+		public void DeleteUnionSubclassAbstractRoot()
+		{
+			if (!Dialect.SupportsTemporaryTables)
+			{
+				Assert.Ignore("Cannot perform multi-table deletes using dialect not supporting temp tables.");
+			}
+
+			// These should reach out into *all* subclass tables...
+			using (var s = OpenSession())
+			using (var t = s.BeginTransaction())
+			{
+				int count = s.Query<Vehicle>().Where(x => x.Owner == "Steve").Delete();
+				Assert.That(count, Is.EqualTo(1), "incorrect restricted update count");
+
+				count = s.Query<Vehicle>().Delete();
+				Assert.That(count, Is.EqualTo(3), "incorrect update count");
+
+				t.Commit();
+				s.Close();
+			}
+		}
+
+		[Test]
+		public void DeleteUnionSubclassConcreteSubclass()
+		{
+			if (!Dialect.SupportsTemporaryTables)
+			{
+				Assert.Ignore("Cannot perform multi-table deletes using dialect not supporting temp tables.");
+			}
+
+			// These should only affect the given table
+			using (var s = OpenSession())
+			using (var t = s.BeginTransaction())
+			{
+				int count = s.Query<Truck>().Where(x => x.Owner == "Steve").Delete();
+				Assert.That(count, Is.EqualTo(1), "incorrect restricted update count");
+
+				count = s.Query<Truck>().Delete();
+				Assert.That(count, Is.EqualTo(2), "incorrect update count");
+				t.Commit();
+				s.Close();
+			}
+		}
+
+		[Test]
+		public void DeleteUnionSubclassLeafSubclass()
+		{
+			// These should only affect the given table
+			using (var s = OpenSession())
+			using (var t = s.BeginTransaction())
+			{
+				int count = s.Query<Car>().Where(x => x.Owner == "Kirsten").Delete();
+				Assert.That(count, Is.EqualTo(1), "incorrect restricted update count");
+
+				count = s.Query<Car>().Delete();
+				Assert.That(count, Is.EqualTo(0), "incorrect update count");
+
+				t.Commit();
+				s.Close();
+			}
+		}
+
+		[Test]
+		public void DeleteRestrictedOnManyToOne()
+		{
+			if (!Dialect.SupportsTemporaryTables)
+			{
+				Assert.Ignore("Cannot perform multi-table deletes using dialect not supporting temp tables.");
+			}
+
+			using (var s = OpenSession())
+			using (var t = s.BeginTransaction())
+			{
+				int count = s.Query<Animal>().Where(x => x.Mother == _data.Butterfly).Delete();
+				Assert.That(count, Is.EqualTo(1));
+
+				t.Commit();
+				s.Close();
+			}
+		}
+
+		[Test]
+		public void DeleteSyntaxWithCompositeId()
+		{
+			using (var s = OpenSession())
+			using (var t = s.BeginTransaction())
+			{
+				s.Query<EntityWithCrazyCompositeKey>().Where(x => x.Id.Id == 1 && x.Id.OtherId == 2).Delete();
+
+				t.Commit();
+				s.Close();
+			}
+		}
+
+		#endregion
+
+		private class TestData
```

Comment:
Sure. That is just the original `BulkManipulation` pattern not completely rewritten. There are a bunch of other tests in this Linq version which still do their own setup/teardown and may get put in normal pattern too.